Feat(Mesh): Migrate built-in routes page (#5071)
Kong/developer.konghq.com: app/mesh/gateway-routes.md

Reviewer: It must be in a different file, but the "Please adjust" lead in before each TF config should just be "Adjust"

---
title: "Configuring built-in routes with MeshHTTPRoute and MeshTCPRoute"
description: "Reference for configuring HTTP and TCP routing through built-in gateways using MeshHTTPRoute and MeshTCPRoute, including hostname matching and weighted backends."
content_type: reference
layout: reference
products:
  - mesh
breadcrumbs:
  - /mesh/
tags:
  - routing
  - service-mesh
  - tcp
min_version:
  mesh: '2.9'
related_resources:
  - text: Set up a built-in gateway
    url: /how-to/set-up-a-built-in-mesh-gateway/
  - text: Built-in gateways
    url: /mesh/built-in-gateway/
  - text: Configuring built-in listeners
    url: /mesh/gateway-listeners/
  - text: MeshHTTPRoute policy
    url: /mesh/policies/meshhttproute/
  - text: MeshTCPRoute policy
    url: /mesh/policies/meshtcproute/
---

To configure how traffic is forwarded from a listener to your mesh services, use [`MeshHTTPRoute`](/mesh/policies/meshhttproute/) and [`MeshTCPRoute`](/mesh/policies/meshtcproute/).

> Using these route resources with a gateway requires using [`spec.targetRef`](/mesh/policies-introduction/) to target gateway data plane proxies.

Reviewer: ```suggestion
You must use [`spec.targetRef`](/mesh/policies-introduction/) to target gateway data plane proxies when you use these route resources with a gateway.
```

### Route hostnames

[`MeshHTTPRoute`](/mesh/policies/meshhttproute/) rules can specify an additional list of hostnames to further limit the traffic handled by those rules. For example:

{% policy_yaml use_meshservice=true %}
```yaml
type: MeshHTTPRoute
name: http-route
mesh: default
spec:
  targetRef:
    kind: MeshGateway
    name: edge-gateway
  to:
    - targetRef:
        kind: Mesh
      rules:
        - matches:
            - path:
                type: PathPrefix
                value: /
          default:
            backendRefs:
              - kind: MeshService
                name: example-v1
                namespace: app
                port: 8080
    - targetRef:
        kind: Mesh
      hostnames:
        - dev.example.com
      rules:
        - matches:
            - path:
                type: PathPrefix
                value: /
          default:
            backendRefs:
              - kind: MeshService
                name: example-v2
                namespace: app
                port: 8080
```
{% endpolicy_yaml %}

> This route would send all traffic to `dev.example.com` to the `v2` backend but other traffic to `v1`.

Reviewer: ```suggestion
This route would send all traffic to `dev.example.com` to the `v2` backend, but other traffic to `v1`.
```
